include/Surelog/SourceCompile/SymbolTable.h

```diff
@@ -50,16 +49,21 @@ class SymbolTable final {
   // Register given "symbol" string as a symbol and return its id.
   // If this is an existing symbol, its ID is returned, otherwise a new one
   // is created.
-  SymbolId registerSymbol(std::string_view symbol);
+  std::pair<SymbolId, const std::string&> add(std::string_view symbol);
```

Comment:
In an effort to reduce std::string and thus have a way in the future to use different memory allocation: wouldn't returning a `std::pair<SymbolId, std::string_view>` be more desirable ?

Our (yours and my) goal is to eventually have `getSymbol(SymbolId)` return a `std::string_view` as well, so introducing now a new API that returns a `const std::string&` makes the work towards that future harder.